examples/controls_example/public/react_controls/data_controls/initialize_data_control.tsx

```diff
@@ -109,15 +109,46 @@ export const initializeDataControl = <EditorState extends object = {}>(
   );
 
   const onEdit = async () => {
-    openDataControlEditor<DefaultDataControlState & EditorState>(
-      { ...stateManager, ...editorStateManager } as ControlStateManager<
-        DefaultDataControlState & EditorState
-      >,
-      controlGroup,
+    // get the initial state from the state manager
+    const mergedStateManager = {
+      ...stateManager,
+      ...editorStateManager,
+    } as ControlStateManager<DefaultDataControlState & EditorState>;
+    const initialState = (
+      Object.keys(mergedStateManager) as Array<keyof DefaultDataControlState & EditorState>
+    ).reduce((prev, key) => {
+      return {
+        ...prev,
+        [key]: mergedStateManager[key]?.getValue(),
+      };
+    }, {} as DefaultDataControlState & EditorState);
+
+    // open the editor to get the new state
+    openDataControlEditor<DefaultDataControlState & EditorState>({
       services,
-      controlType,
-      controlId
-    );
+      onSave: ({ type: newType, state: newState }) => {
+        if (newType === controlType) {
+          // apply the changes from the new state via the state manager
+          (Object.keys(initialState) as Array<keyof DefaultDataControlState & EditorState>).forEach(
+            (key) => {
+              if (mergedStateManager[key].getValue() !== newState[key]) {
```

Comment:
What do you think about using lodash `isEqual`? There is nothing that enforces that keyed values are primitives. This comparison will not support objects or arrays.

Reply:
🤦 Good call - done in b5147af6b74fdab2146ab21520ec611c7f0bde6a